plugin/manager_linux.go

```diff
@@ -77,14 +78,37 @@ func (pm *Manager) enable(p *v2.Plugin, c *controller, force bool) error {
 }
 
 func (pm *Manager) pluginPostStart(p *v2.Plugin, c *controller) error {
-	client, err := plugins.NewClientWithTimeout("unix://"+filepath.Join(pm.config.ExecRoot, p.GetID(), p.GetSocket()), nil, c.timeoutInSecs)
+	sockAddr := filepath.Join(pm.config.ExecRoot, p.GetID(), p.GetSocket())
+	client, err := plugins.NewClientWithTimeout("unix://"+sockAddr, nil, c.timeoutInSecs)
 	if err != nil {
 		c.restart = false
 		shutdownPlugin(p, c, pm.containerdClient)
 		return errors.WithStack(err)
 	}
 
 	p.SetPClient(client)
+
+	maxRetries := 3
+	var retries int
+	var conn net.Conn
+	for {
+		time.Sleep(3 * time.Second)
+		retries++
+
+		if retries > maxRetries {
+			logrus.Debugf("error net dialing plugin: %v", err)
+			c.restart = false
+			shutdownPlugin(p, c, pm.containerdClient)
+			return err
+		}
+
+		// net dial into the unix socket to see if someone's listening.
+		conn, err = net.Dial("unix", sockAddr)
+		if err == nil {
+			break
+		}
+	}
+	conn.Close()
```

Comment:
Done moving `Close`. Dont see a big need for moving to another tiny function, since it fits in `postPluginStart` nicely.